src/main/Store.java

```diff
@@ -0,0 +1,131 @@
+package main;
+
+/**
+ * General class that keeps actions of the store
+ *
+ * @author Radu Popescu
+ */
+
+import product.AnimalProduct;
+import product.Product;
+import product.VegetalProduct;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class Store {
+    private final String[] animalProductNames = new String[]{"milk", "eggs", "cheese", "yogurt", "sour cream"};
+    public List stockItemList = new ArrayList();
+    public List soldItemList = new ArrayList();
+
+    public int readInt(String intMessageRequest) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println(intMessageRequest);
+        int value = sc.nextInt();
+        return value;
+    }
+
+    public String readString(String stringMessageRequest) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println(stringMessageRequest);
+        String text = sc.nextLine();
+        return text;
+    }
+
+    public Product addProduct(String name, int price, String date, int weight) throws ParseException {
+
+        SimpleDateFormat format = new SimpleDateFormat("yyyy-mm-dd");
+        Date validityDate = format.parse(date);
+
+        int id = stockItemList.size() + 1;
+
+        return new Product(name, price, validityDate, weight, id);
+    }
+
+    public void addAnimalProduct(String name, int price, String date, int weight, int storageTemperature, int quantity) throws ParseException {
+
+        if (Arrays.asList(animalProductNames).contains(name)) {
+
+            Product tmpProduct = addProduct(name, price, date, weight);
```

Comment:
you're creating a bit of waste here creating this tmpProduct just to use it to add stuff to animalProduct.

I would delete the addProduct method, I would put the SimpleDateFormat format as a constant on my class (because I'm never changing the format and I can use the same instance multiple times to parse dates).

Then my animal product would be created like this:
new AnimalProduct(name, price, format.parse(date), weight, stockItemList.size()+1, storageTemperature);